src/map_extras.cpp

```diff
@@ -1399,17 +1453,130 @@ static void mx_fumarole( map &m, const tripoint &abs_sub )
+                        m.adjust_radiation( loc.x, loc.y, rng( 20, 40 ) );
+                    }
+                }
+            }
+            break;
+        }
+        //Lava seams originating from the portal
+        case 5: {
+            if( abs_sub.z <= 0 ) {
+                int x1 = rng( 0,    SEEX     - 3 ), y1 = rng( 0,    SEEY     - 3 ),
+                    x2 = rng( SEEX, SEEX * 2 - 3 ), y2 = rng( SEEY, SEEY * 2 - 3 );
+
+                // Pick a random cardinal direction to also spawn lava in
+                // This will make the lava a single connected line, not just on diagonals
+                std::vector<direction> possibilities;
+                possibilities.push_back( EAST );
+                possibilities.push_back( WEST );
+                possibilities.push_back( NORTH );
+                possibilities.push_back( SOUTH );
+                const direction extra_lava_dir = random_entry( possibilities );
+                int x_extra = 0;
+                int y_extra = 0;
+                switch( extra_lava_dir ) {
+                    case NORTH:
+                        y_extra = -1;
+                        break;
+                    case EAST:
+                        x_extra = 1;
+                        break;
+                    case SOUTH:
+                        y_extra = 1;
+                        break;
+                    case WEST:
+                        x_extra = -1;
+                        break;
+                    default:
+                        break;
+                }
+
+                const tripoint portal_location = { x1 + x_extra, y1 + y_extra, abs_sub.z };
+                m.add_field( portal_location, fd_fatigue, 3 );
+
+                std::set<point> ignited;
+                place_fumarole( m, x1, y1, x2, y2, ignited );
+                place_fumarole( m, x1 + x_extra, y1 + y_extra, x2 + x_extra, y2 + y_extra, ignited );
+
+                for( auto &i : ignited ) {
+                    // Don't need to do anything to tiles that already have lava on them
+                    if( m.ter( i.x, i.y ) != t_lava ) {
+                        // Spawn an intense but short-lived fire
+                        // Any furniture or buildings will catch fire, otherwise it will burn out quickly
+                        m.add_field( tripoint( i.x, i.y, abs_sub.z ), fd_fire, 15, 1_minutes );
+                    }
+                }
+            }
+            break;
+        }
+        case 6: {
+            //Mi-go went through the portal and began constructing their base of operations
+            m.add_field( portal_location, fd_fatigue, 3 );
+            for( int i = x - 5; i <= x + 5; i++ ) {
+                for( int j = y - 5; j <= y + 5; j++ ) {
+                    if( rng( 1, 9 ) >= trig_dist( x, y, i, j ) ) {
+                        if( one_in( 15 ) ) {
+                            m.place_spawns( GROUP_MI_GO_CAMP_OM, 1, i, j, i, j, 1, true );
```

Comment:
Why not leave the placing to `place_spawns`? It allows specifying and area instead of a single point. This would also ensure the creatures don't have to spawn on impassable terrain.

Something like:
```suggestion
                            m.place_spawns( GROUP_MI_GO_CAMP_OM, 1, x-5, y-5, x+5, y+5, 1, true );
```
(You may have to increase the chance / density.)

Reply:
Fixed.